packages/lit-dev-content/site/docs/frameworks/react.md

```diff
@@ -0,0 +1,153 @@
+---
+title: React
+eleventyNavigation:
+  key: React
+  parent: Framework integration
+  order: 1
+  labs: true
+---
+
+{% labs-disclaimer %}
+
+The [@lit-labs/react](https://github.com/lit/lit/tree/main/packages/labs/react) package provides web component and reactive controller integration for React. While React can render web components as-is, there are a few gaps in the developer experience. You can't easily pass React props to custom element properties or add event listeners to custom elements. (For more information on the limitations of React's web component integration, see [Custom Elements Everywhere](https://custom-elements-everywhere.com/libraries/react/results/results.html).)
+
+The `@lit-labs/react` package provides two main entry points:
+
+-   `createComponent()` creates a React component that _wraps_ an existing web component. The wrapper allows you to set props on the component and add event listeners to the component like you would any other React component.
+
+-   `useController()` lets you use a Lit reactive controller as a React hook.  
+
+## createComponent
+
+The `createComponent()` function makes a React component wrapper for a custom element class. The wrapper correctly passes React `props` to properties accepted by the custom element and listens for events dispatched by the custom element.
+
+### How it works
+
+For properties, the wrapper interrogates the web component class to discover its available properties. Then any React props passed with property names are set on the custom element as properties and not attributes.
+
+For events, `createComponent()` accepts a mapping of React event prop names to events fired by the custom element. For example passing `{onFoo: 'foo'}` means a function passed via a prop named `onFoo` will be called when the custom element fires the `foo` event with the event as an argument.
+
+### Usage
+
+Import `React`, a custom element class, and `createComponent`.
+
+```js
+import React from 'react';
+import {createComponent} from '@lit-labs/react';
+import {MyElement} from './my-element.js';
+
+export const MyElementComponent = createComponent({
+  tagName: 'my-element',
+  elementClass: MyElement,
+  react: React,
+  events: {
+    onActivate: 'activate',
+    onChange: 'change',
+  },
+});
+```
+
+After defining the React component, you can use it just as you would any other React component.
+
+```jsx
+<MyElementComponent
+  active={isActive}
+  onActivate={(e) => setIsActive(e.active)}
+/>
+```
+
+#### Typescript
```

Comment:
I'm not sure a separate TypeScript section makes as much sense as having details on properties and events. This whole section is basically on events.

A properties section could talk about the difference between properties and attributes when you have both?

I also think it might be worth covering each option: tagName (must be the name the element was registered with), elementClass (hopefully self-explanatory?), react (for compatibility with CJS and so you could pass in Preact).

Reply:
I added a section for the options preceding this. I renamed the "Typescript" heading to "Typing event handler props" as a usage example of the `EventName` utility type we vend. We go into properties/attributes in the "how it works" section.